spec/01-unit/01-db/01-schema/01-schema_spec.lua

```diff
@@ -2692,6 +2692,54 @@ describe("schema", function()
       assert.same(ngx.null, data.i)
     end)
 
+    it("produces nil for empty string fields with selects", function()
+      local Test = Schema.new({
+        fields = {
+          { a = { type = "string" }, },
+          { b = { type = "record", fields = { { c = { type = "string" }, } } } },
+          { d = { type = "array", elements = { type = "string" } }, },
+          { e = { type = "string", len_min = 0 }, },
+        }
+      })
+      local data, err = Test:process_auto_fields({
+        a = "",
+        b = {
+          c = "",
+        },
+        d = { "", "", "" },
+        e = "",
+      }, "select")
+      assert.is_nil(err)
+      assert.equal(nil, data.a)
+      assert.equal(nil, data.b.c)
+      assert.same({"", "", ""}, data.d) -- TODO: should we remove also empty values from arrays?
+      assert.equal("", data.e)
+    end)
+
+    it("produces ngx.null (when asked) for empty string fields with selects", function()
+      local Test = Schema.new({
+        fields = {
+          { a = { type = "string" }, },
+          { b = { type = "record", fields = { { c = { type = "string" }, } } } },
+          { d = { type = "array", elements = { type = "string" } }, },
+          { e = { type = "string", len_min = 0 }, },
+        }
+      })
+      local data, err = Test:process_auto_fields({
+        a = "",
+        b = {
+          c = "",
+        },
+        d = { "", "", "" },
+        e = "",
+      }, "select", true)
+      assert.is_nil(err)
+      assert.equal(cjson.null, data.a)
+      assert.equal(cjson.null, data.b.c)
+      assert.same({"", "", ""}, data.d) -- TODO: should we remove also empty values from arrays?
```

Comment:
Good question. To reiterate, in order to fix https://github.com/Kong/kong/issues/5653 we just need to remove empty strings. But what about official or third party plugins that have an array of empty strings in the database? Or any other type that accepts elements of strings, for example, a map or a set? See all the types at https://github.com/Kong/kong/blob/master/kong/db/schema/init.lua#L52-L60

Reply:
I think we should do the same treatment for anything that accepts strings as element, no? We don't allow empty strings elements inside arrays. So If we also have it set to "" in empty in the DB, shouldn't we also translate it to `nil`?

Reply:
For maps that have `string` values (or records as values that have `strings`) I think, absolutely, that would be a good idea. Arrays/sets I am not sure what the current state is. When removing empty string from arrays/set we need to to use `table.remove` to prevent holes.

I added a couple of tests there, so if we modify the behavior those tests should fail. I think it might be a good idea to do the same treatment. @hishamhm do you have anything against it? Is it part of this PR or should it be another PR?